### PR TITLE
Add transparent init

### LINKED ISSUE #1273: Transparent `skaffold init`
When no skaffold configuration is found on disk, maybe we could run `skaffold init` in memory and use that.
I think it would be a very simple way of getting started with Skaffold and also a good way of supporting things like compose files transparently.
We’d, of course, print a warning to explain what’s going on and which configuration Skaffold is using.

### PR DESCRIPTION
**Pull request by MarlonGamez**
Fixes #1273 
**Related** #4915, #5037, #5141, #5155

**Description**
Adds the flow that handles users running commands without a skaffold config file being present. 

**User facing changes (remove if N/A)**
If users try to run a command without a skaffold config such as `skaffold dev`, skaffold will tell them that no config is found, and that it is trying to make one. If it succeeds, it will prompt the user, making sure that they want to continue.

### CODE REVIEW

File: cmd/skaffold/app/cmd/runner.go [hunk 1]
```diff
@@ -71,11 +75,58 @@ func createNewRunner(opts config.SkaffoldOptions) (runner.Runner, []*latest.Skaf
...
 	if err != nil {
 		if os.IsNotExist(errors.Unwrap(err)) {
-			return nil, nil, fmt.Errorf("skaffold config file %s not found - check your current working directory, or try running `skaffold init`", opts.ConfigurationFile)
+			color.Default.Fprintf(out, "Skaffold config file %s not found - Trying to create one for you...\n", opts.ConfigurationFile)
+			config, err := initializer.Transparent(context.Background(), out, initConfig.Config{Opts: opts})
+			if err != nil {
+				return nil, nil, fmt.Errorf("unable to generate skaffold config file automatically - try running `skaffold init`: %w", err)
+			}
+			if config == nil {
+				return nil, nil, fmt.Errorf("user not continuing")
```
Comment (gsquared94) [inline cmd/skaffold/app/cmd/runner.go:124]: should this be a bit more verbose?

**Comment by gsquared94** [comment]
looks good. Should we add two flags:
- one to skip the confirmation `? Would you like to continue? Yes`
- one to skip this feature and fail as before

File: cmd/skaffold/app/cmd/runner.go (hunk 1 above)
Comment (MarlonGamez) [reply cmd/skaffold/app/cmd/runner.go:124]: Is there anything specific you feel should be mentioned here? Maybe something like
> user not continuing with generated config

Comment (gsquared94) [reply cmd/skaffold/app/cmd/runner.go:124]: I was thinking something more fatal like:
```go
return nil, nil, errors.New("unable to generate skaffold config file automatically - try running `skaffold init`: action cancelled by user")
```

File: cmd/skaffold/app/cmd/flags.go [hunk 2]
```diff
@@ -460,6 +460,22 @@ var flagRegistry = []Flag{
 		FlagAddMethod: "Var",
 		DefinedOn:     []string{"test", "deploy"},
 	},
+	{
+		Name:          "try-transparent-init",
```
Comment (gsquared94) [inline cmd/skaffold/app/cmd/flags.go:464]: `auto-create-config`?

File: cmd/skaffold/app/cmd/flags.go [hunk 3]
```diff
@@ -460,6 +460,22 @@ var flagRegistry = []Flag{
...
+		Usage:         "If true, skaffold will try to create a config for the user's run if it doesn't find one",
+		Value:         &opts.TryTransparentInit,
+		DefValue:      true,
+		FlagAddMethod: "BoolVar",
+		DefinedOn:     []string{"debug", "dev", "run"},
+	},
+	{
+		Name:          "skip-confirmation",
```
Comment (gsquared94) [inline cmd/skaffold/app/cmd/flags.go:472]: `--assume-yes`? also will this be globally applied to skip prompt from all user interactions?

File: cmd/skaffold/app/cmd/runner.go (hunk 1 above)
Comment (MarlonGamez) [reply cmd/skaffold/app/cmd/runner.go]: makes sense, I can change that

File: cmd/skaffold/app/cmd/flags.go (hunk 2 above)
Comment (MarlonGamez) [reply cmd/skaffold/app/cmd/flags.go]: done

File: cmd/skaffold/app/cmd/flags.go (hunk 3 above)
Comment (MarlonGamez) [reply cmd/skaffold/app/cmd/flags.go]: done

File: cmd/skaffold/app/cmd/flags.go [hunk 4]
```diff
@@ -460,6 +460,22 @@ var flagRegistry = []Flag{
 		FlagAddMethod: "Var",
 		DefinedOn:     []string{"test", "deploy"},
 	},
+	{
+		Name:          "auto-create-config",
```
Comment (IsaacPD) [inline cmd/skaffold/app/cmd/flags.go:464]: Could you set the `IsEnum` field to `true` for this flag and `assume-yes`?

Comment (MarlonGamez) [reply cmd/skaffold/app/cmd/flags.go:464]: yep, done

**Review by IsaacPD — APPROVED** [review]

**Review by gsquared94 — APPROVED** [review]
lgtm!

**Pull request merged**